content/doc/developer/internationalization/prepare-a-java-source-file.adoc

```diff
@@ -0,0 +1,118 @@
+---
+title: Prepare a Java Source File for Localization
+layout: developerguide
+references:
+- url: https://en.wikipedia.org/wiki/.properties
+  title: .properties file format on Wikipedia
+---
+:toc:
+
+This guide explains how to prepare a Java source file in Jenkins core or a plugin for localization.
+
+== Step 1: Identify messages
+
+First, we need to identify the localizable strings that should be translated. In general, everything shown on any user interface should be internationalized (i.e. prepared for localization into the user's language).
+
+Historically, log messages to be written to the Jenkins server log file from a logger aren't translated, but could also be.
+
+Let's consider the following example, taken from the sample +HelloWorldBuilder+:
```

Comment:
Why

    +foo+

not

    `foo`

? GitHub preview does not grok the former.